xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -1863,28 +1884,23 @@ private String userReferenceToString(DocumentReference userReference)
 
     /**
      * @since 3.0M3
+     * @deprecated Since 13.10RC1 rely on {@link #getAuthors}.
      */
+    @Deprecated
     public DocumentReference getAuthorReference()
     {
-        return this.authorReference;
+        return this.getDocumentReferenceUserReferenceResolver().resolve(getAuthors().getMetadataAuthor());
     }
 
     /**
      * @since 3.0M3
+     * @deprecated Since 13.10RC1 rely on {@link #setAuthors(DocumentAuthors)}
      */
+    @Deprecated
     public void setAuthorReference(DocumentReference authorReference)
     {
-        if (ObjectUtils.notEqual(authorReference, getAuthorReference())) {
-            setMetaDataDirty(true);
-        }
-
-        this.authorReference = intern(authorReference);
-
-        // Log this since it's probably a mistake so that we find who is doing bad things
-        if (this.authorReference != null && this.authorReference.getName().equals(XWikiRightService.GUEST_USER)) {
-            LOGGER.warn("A reference to XWikiGuest user has been set instead of null. This is probably a mistake.",
-                new Exception("See stack trace"));
-        }
+        UserReference user = this.getUserReferenceDocumentReferenceResolver().resolve(authorReference);
+        this.setAuthors(new DefaultDocumentAuthors(this.getAuthors()).setMetadataAuthor(user));
```

Comment:
Why creating a new DefaultDocumentAuthors for each set instead of just modifying the existing one ?

Reply:
Because XWikiDocument contains the interface DocumentAuthor, not the DefaultDocumentAuthor: in the future we can imagine that there's others implementations, and it would still work.

Reply:
Except that as soon as a set is called you loose the previous custom one and replace it with DefaultDocumentAuthor so you might set DefaultDocumentAuthor from the start and just call set on it.

Reply:
By the way not exposing any set in DocumentAuthors means that changing one of those authors is quite a pain and close to impossible for an extension without using deprecated APIs since DefaultDocumentAuthors is internal.

Reply:
Right.